src/contexts/states/viewStore.ts

```diff
@@ -0,0 +1,246 @@
+                useViewStore.getState().updatePageData(pageData);
+            })
+            .catch((e: unknown) => {
+                useContextStore.getState().postPopUp({
+                    level: LOG_LEVEL.ERROR,
+                    message: String(e),
+                    timeoutMillis: DO_NOT_TIMEOUT_VALUE,
+                    title: "Action failed",
+                });
+            });
+        useQueryStore.getState().startQuery();
+    },
+    loadPageByAction: (navAction: NavigationAction) => {
+        const {logEventNum} = useContextStore.getState();
+        const {fileSrc, loadFile} = useLogFileStore.getState();
+        if (navAction.code === ACTION_NAME.RELOAD) {
+            if (null === fileSrc || 0 === logEventNum) {
+                throw new Error(
+                    `Unexpected fileSrc=${JSON.stringify(
+                        fileSrc
+                    )}, logEventNum=${logEventNum} when reloading.`
+                );
+            }
+            loadFile(fileSrc, {
+                code: CURSOR_CODE.EVENT_NUM,
+                args: {eventNum: logEventNum},
+            });
+
+            return;
+        }
+
+        const {isPrettified, numPages, pageNum} = get();
+        const cursor = getPageNumCursor(navAction, pageNum, numPages);
+        if (null === cursor) {
+            console.error(`Error with nav action ${navAction.code}.`);
+
+            return;
+        }
+
+        useUiStore.getState().setUiState(UI_STATE.FAST_LOADING);
+
+        useLogFileManagerStore.getState().logFileManagerProxy.loadPage(
+            cursor,
+            isPrettified
+        ).then((pageData: PageData) => {
+            useViewStore.getState().updatePageData(pageData);
+        })
+            .catch((e: unknown) => {
+                useContextStore.getState().postPopUp({
+                    level: LOG_LEVEL.ERROR,
+                    message: String(e),
+                    timeoutMillis: DO_NOT_TIMEOUT_VALUE,
+                    title: "Action failed",
+                });
+            });
+    },
+    setBeginLineNumToLogEventNum: (newMap) => {
+        set({beginLineNumToLogEventNum: newMap});
+    },
+    setIsPrettified: (newIsPrettified: boolean) => {
+        if (newIsPrettified === get().isPrettified) {
+            return;
+        }
+        set({isPrettified: newIsPrettified});
+        useUiStore.getState().setUiState(UI_STATE.FAST_LOADING);
+
+        const {logEventNum} = useContextStore.getState();
+        let cursor: CursorType = {code: CURSOR_CODE.LAST_EVENT, args: null};
+        if (0 !== logEventNum) {
+            cursor = {
+                code: CURSOR_CODE.EVENT_NUM,
+                args: {eventNum: logEventNum},
+            };
+        }
+
+        useLogFileManagerStore
+            .getState()
+            .logFileManagerProxy
+            .loadPage(
+                cursor,
+                newIsPrettified
+            ).then((pageData: PageData) => {
```

Comment:
ditto: change `setIsPrettified` as async function or use the inline async function syntax we discussed yesterday